backend/packages/Upgrade/src/api/controllers/ExperimentClientController.v4.ts

```diff
@@ -0,0 +1,937 @@
+    const experimentUserDoc = await this.getUserDoc(workingGroupParams.id, request.logger);
+    if (experimentUserDoc) {
+      // append userDoc in logger
+      request.logger.child({ userDoc: experimentUserDoc });
+      request.logger.info({ message: 'Got the original user doc' });
+    }
+    const { id, workingGroup } = await this.experimentUserService.updateWorkingGroup(
+      workingGroupParams.id,
+      workingGroupParams.workingGroup,
+      {
+        logger: request.logger,
+        userDoc: experimentUserDoc,
+      }
+    );
+    return { id, workingGroup };
+  }
+
+  /**
+   * @swagger
+   * /mark:
+   *    post:
+   *       description: Mark a Experiment Point
+   *       consumes:
+   *         - application/json
+   *       parameters:
+   *         - in: body
+   *           name: experimentUser
+   *           required: true
+   *           schema:
+   *             type: object
+   *             required:
+   *                - userId
+   *                - experimentPoint
+   *                - condition
+   *             properties:
+   *               userId:
+   *                 type: string
+   *               site:
+   *                 type: string
+   *               target:
+   *                 type: string
+   *                 example: partition1
+   *               condition:
+   *                 type: string
+   *                 example: control
+   *               status:
+   *                 type: string
+   *                 example: condition applied
+   *               experimentId:
+   *                 type: string
+   *                 example: exp1
+   *           description: ExperimentUser
+   *       tags:
+   *         - Client Side SDK
+   *       produces:
+   *         - application/json
+   *       responses:
+   *          '200':
+   *            description: Experiment Point is Marked
+   *            schema:
+   *              type: object
+   *              properties:
+   *                id:
+   *                  type: string
+   *                  minLength: 1
+   *                experimentId:
+   *                  type: string
+   *                  minLength: 1
+   *                site:
+   *                  type: string
+   *                  minLength: 1
+   *                target:
+   *                  type: string
+   *                  minLength: 1
+   *                condition:
+   *                  type: string
+   *                  minLength: 1
+   *              required:
+   *                - id
+   *                - experimentId
+   *                - enrollmentCode
+   *                - userId
+   *                - condition
+   *          '500':
+   *            description: User not defined
+   */
+  @Post('mark')
+  public async markExperimentPoint(
+    @Body({ validate: { validationError: { target: false, value: false } } })
+    @Req()
+    request: AppRequest,
+    experiment: MarkExperimentValidator2
+  ): Promise<IMonitoredDeciosionPoint> {
```

Comment:
Change spelling error `IMonitoredDeciosionPoint` to `IMonitoredDecisionPoint`

Reply:
Code updated